values/binary.go

```diff
@@ -53,128 +53,123 @@ var binaryFuncLookup = map[BinaryFuncSignature]BinaryFunction{
 		l := lv.UInt()
 		r := rv.UInt()
-		return NewUInt(l - r)
+		return NewUInt(l - r), nil
 	},
-	{Operator: ast.SubtractionOperator, Left: semantic.Float, Right: semantic.Float}: func(lv, rv Value) Value {
+	{Operator: ast.SubtractionOperator, Left: semantic.Float, Right: semantic.Float}: func(lv, rv Value) (Value, error) {
 		l := lv.Float()
 		r := rv.Float()
-		return NewFloat(l - r)
+		return NewFloat(l - r), nil
 	},
 	{Operator: ast.SubtractionOperator, Left: semantic.Nil, Right: semantic.Nil}: nil,
-	{Operator: ast.MultiplicationOperator, Left: semantic.Int, Right: semantic.Int}: func(lv, rv Value) Value {
+	{Operator: ast.MultiplicationOperator, Left: semantic.Int, Right: semantic.Int}: func(lv, rv Value) (Value, error) {
 		l := lv.Int()
 		r := rv.Int()
-		return NewInt(l * r)
+		return NewInt(l * r), nil
 	},
-	{Operator: ast.MultiplicationOperator, Left: semantic.UInt, Right: semantic.UInt}: func(lv, rv Value) Value {
+	{Operator: ast.MultiplicationOperator, Left: semantic.UInt, Right: semantic.UInt}: func(lv, rv Value) (Value, error) {
 		l := lv.UInt()
 		r := rv.UInt()
-		return NewUInt(l * r)
+		return NewUInt(l * r), nil
 	},
-	{Operator: ast.MultiplicationOperator, Left: semantic.Float, Right: semantic.Float}: func(lv, rv Value) Value {
+	{Operator: ast.MultiplicationOperator, Left: semantic.Float, Right: semantic.Float}: func(lv, rv Value) (Value, error) {
 		l := lv.Float()
 		r := rv.Float()
-		return NewFloat(l * r)
+		return NewFloat(l * r), nil
 	},
 	{Operator: ast.MultiplicationOperator, Left: semantic.Nil, Right: semantic.Nil}: nil,
-	{Operator: ast.DivisionOperator, Left: semantic.Int, Right: semantic.Int}: func(lv, rv Value) Value {
+	{Operator: ast.DivisionOperator, Left: semantic.Int, Right: semantic.Int}: func(lv, rv Value) (Value, error) {
 		l := lv.Int()
 		r := rv.Int()
 		if r == 0 {
-			// TODO(#38): reject divisions with a constant 0 divisor.
-			return NewInt(0)
+			return nil, fmt.Errorf("cannot divide by zero")
 		}
-		return NewInt(l / r)
+		return NewInt(l / r), nil
 	},
-	{Operator: ast.DivisionOperator, Left: semantic.UInt, Right: semantic.UInt}: func(lv, rv Value) Value {
+	{Operator: ast.DivisionOperator, Left: semantic.UInt, Right: semantic.UInt}: func(lv, rv Value) (Value, error) {
 		l := lv.UInt()
 		r := rv.UInt()
 		if r == 0 {
-			// TODO(#38): reject divisions with a constant 0 divisor.
-			return NewUInt(0)
+			return nil, fmt.Errorf("cannot divide by zero")
 		}
-		return NewUInt(l / r)
+		return NewUInt(l / r), nil
 	},
-	{Operator: ast.DivisionOperator, Left: semantic.Float, Right: semantic.Float}: func(lv, rv Value) Value {
+	{Operator: ast.DivisionOperator, Left: semantic.Float, Right: semantic.Float}: func(lv, rv Value) (Value, error) {
 		l := lv.Float()
 		r := rv.Float()
 		if r == 0 {
-			// TODO(#38): reject divisions with a constant 0 divisor.
-			return NewFloat(math.NaN())
+			return nil, fmt.Errorf("cannot divide by zero")
```

Comment:
Is this what we want to do for floating point division?  It seems like `NaN` is a reasonable thing to return.  Do any of the issues related to division make a distinction between float and integer division?

Reply:
#Flux concluded that throwing an error is the best option to avoid the confusion of IEEE754. Marking as resolved.